### PR TITLE
Remove all calls to goog.math

### PR DESCRIPTION
**Pull request by samelhusseini**
Remove calls to goog.math.
It turns out we're only using goog.math.Size and goog.math.Coordinate and we're using them as a struct and not utilizing any of their helper methods.
Unlike https://github.com/google/blockly/pull/2108, the proposal here is to keep the structs the way they are and just reference our version instead of closure's. 
This way the shape doesn't change and won't break anyone depending on these fields having certain properties.

### CODE REVIEW

File: core/utils/math.js [hunk 1]
```diff
@@ -68,3 +68,47 @@ Blockly.utils.math.clamp = function(lowerBound, number, upperBound) {
...
+ * @param {number=} opt_y Top, defaults to 0.
+ * @struct
+ * @constructor
+ */
+Blockly.utils.math.Coordinate = function(opt_x, opt_y) {
+  /**
+   * X-value
+   * @type {number}
+   */
+  this.x = goog.isDef(opt_x) ? opt_x : 0;
```
Comment (NeilFraser) [inline core/utils/math.js]: Please don't introduce new dependencies on Closure.

**Comment by samelhusseini** [comment]
No difference, my bad looks like that was already defined.  Fixed.

File: core/utils/size.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Utility methods for size calculation.
+ * These methods are not specific to Blockly, and could be factored out into
+ * a JavaScript framework such as Closure.
+ * @author fraser@google.com (Neil Fraser)
```
Comment (NeilFraser) [inline core/utils/size.js]: Are you the new me?

File: core/gesture.js [hunk 3]
```diff
@@ -764,7 +764,7 @@ Blockly.Gesture.prototype.doWorkspaceClick_ = function(e) {
   var ws = this.creatorWorkspace_;
   ws.cursor.hide();
   if (e.shiftKey) {
-    var screenCoord = new goog.math.Coordinate(e.clientX, e.clientY);
```
Comment (NeilFraser) [inline core/gesture.js]: I'd love to know how this one slipped through my efforts last month...
Ah, @alschmiedt added it last week.  It's not in master.

File: tests/jsunit/event_test.js [hunk 4]
```diff
@@ -238,7 +238,7 @@ function test_move_constructorCoordinate() {
   setUpMockMethod(mockControl_, Blockly.utils, 'genUid', null, ['1', '2']);
   try {
     var block1 = createSimpleTestBlock(workspace);
-    var coordinate = new goog.math.Coordinate(3,4);
+    var coordinate = new Blockly.utils.Coordinate(3,4);
```
Comment (NeilFraser) [inline tests/jsunit/event_test.js]: Not the fault of this PR, but can we get some spaces between arguments here and below?

**Pull request merged**